metricbeat/metricbeat.reference.yml

```diff
@@ -351,6 +351,7 @@ metricbeat.modules:
   metricsets: ["status"]
   period: 10s
   hosts: ["localhost:5601"]
+  basepath: ""
```

Comment:
Is the basepath mainly used for development? Would it also work the specify the host as `localhost:5601/abc`

Reply:
Basepath gets used in production scenarios too, where users deploy Kibana and also other applications behind the same proxy and need a unique base path to route to each application.

Reply:
Ok, then I think a separate config makes sense. I assume including in the host would also work?

Reply:
>  I assume including in the host would also work?

You mean something like `hosts: ["localhost:5601/foo"]` in `modules.d/kibana.yml`, right?

Heh, I swear I tried that first and it didn't work at the time, so I went with a separate setting. However, I just tried it again, and it works! So now I think we should get rid of the extra `basepath` setting.

Reply:
Ugh, spoke too soon. It doesn't work when I add it to the `hosts` setting. 😞 

What seems to happen is that the URL becomes `http://localhost:5601/foo?extended=true` instead of `http://localhost:5601/foo/api/stats?extended=true`.

So now I'm back to thinking we should give `basepath` its own setting.

Reply:
BTW, even if we decide to have a separate `basepath` setting, I think there's a better implementation for utilizing it. I've made this alternate implementation in 1379dd89acd21cc2e7be23570e20f5c27891d5fa. Let me know what you think of it.

Reply:
I like the idea of having the basepath in our http parser directly as I remember also other modules use it. Not 100% about the name basepath or something like prefix at the same time does not matter much.

@andrewkroh WDYT about having basepath as part of the http parser?

Reply:
I would prefer to see if we can change the behavior of the http parser such that if path is included in the `hosts` URL that it still does the right thing. It seems like that was the expected behavior. Fewer config options and an explicit URL sounds simpler to me, but I may not have the whole picture.

> What seems to happen is that the URL becomes http://localhost:5601/foo?extended=true instead of http://localhost:5601/foo/api/stats?extended=true.

Reply:
That makes sense @andrewkroh. Thanks for weighing in. I'll give this a shot in a new PR this coming week.

Reply:
@ycombinator Could you set it a priority so we get it into 6.4?

Reply:
Yep, I plan to have a PR up today!

Reply:
PR is up and I've requested a review from both of you: https://github.com/elastic/beats/pull/7682